Oae 0db5204

katarkow opened the pull request:
## Purpose of this PR


## Type of change

(Make sure to delete from the Type-of-change list the items not relevant to your PR)

- [ ] Bug fix 
- [ ] Refactoring
- [ ] New feature 
- [ ] Minor change (default scenarios show only small differences)
- [ ] Fundamental change
- [ ] This change requires a documentation update


## Checklist:

- [ ] My code follows the [coding etiquette](https://github.com/remindmodel/remind/blob/develop/main.gms#L80)
- [ ] I performed a self-review of my own code
- [ ] I explained my changes within the PR, particularly in hard-to-understand areas
- [ ] I checked that the [in-code documentation](https://github.com/remindmodel/remind/blob/develop/main.gms#L120) is up-to-date
- [ ] I adjusted the reporting in [`remind2`](https://github.com/pik-piam/remind2) where it was needed
- [ ] I adjusted `forbiddenColumnNames` in [readCheckScenarioConfig.R](https://github.com/remindmodel/remind/blob/develop/scripts/start/readCheckScenarioConfig.R) in case the PR leads to deprecated switches
- [ ] All automated model tests pass (`FAIL 0` in the output of `make test`)
- [ ] The changelog `CHANGELOG.md` [has been updated correctly](https://gitlab.pik-potsdam.de/rse/rsewiki/-/wikis/Standards-for-Writing-a-Changelog)

## Further information (optional):

* Test runs are here: 
* Comparison of results (what changes by this PR?):

katarkow committed:
Add OAE to the CDR portfolio

Commit:
only start industry carbon capture in Germany by 2030 as no significant projects before in pipeline

katarkow committed:
Add qttyTarget to OAE

Commit:
add bounds to align CHP capacities in 2020 for Germany with historic values

Commit:
add bounds for aligning historic and near-term power sector capacities with historic data and add switch cm_tech_bounds_2025 that adds generous bounds for 2025 of spv and wind given recent developments. The switch is activated by default and can be used for further 2025 bounds of that kind that may want to be deactivated for certain purposes.

Commit:
add ariadne scenario config

katarkow committed:
Update paths to remind2

katarkow committed:
Update OAE costs

katarkow committed:
Update parameters, matching to OAE NG, lower omf

Commit:
some renamings for codeCheck

Commit:
update ariadne scenario config

Commit:
delete some outdated columns  ined columns in ariadne scenario config

katarkow committed:
Split into natural gas calciner and novel calciner

strefler commented on main.gms:
```diff
@@ -848,6 +848,32 @@ parameter
+*' * (0): not included
+*'
+parameter
+  cm_33_OAE_eff             "OAE efficiency measured in tCO2 uptaken by the ocean per tCaO. Typically between 0.9-1.4. [tCO2/tCaO]"
```
would be great to add moles as well for comparison

strefler commented on main.gms:
```diff
@@ -848,6 +848,32 @@ parameter
+parameter
+  cm_33_OAE_startyr         "The year when OAE could start being deployed [year]"
+;
+  cm_33_OAE_startyr        = 2025; !! def = 2025  !! regexp = 2025|20[3-9](0|5)
```
change to 2030

strefler commented on main.gms:
```diff
@@ -1298,7 +1324,7 @@ $setGlobal cm_proNucRegiPol   off   !! def = off
 *** cm_vehiclesSubsidies - If "on" applies country specific BEV and FCEV subsidies from 2020 onwards
 $setGlobal cm_vehiclesSubsidies  off !! def = off
-*** cm_implicitQttyTarget - Define quantity target for primary, secondary, final energy or CCS (PE, SE and FE in TWa, or CCS in Mt CO2) per target group (total, biomass, fossil, VRE, renewables, synthetic, ...).
+*** cm_implicitQttyTarget - Define quantity target for primary, secondary, final energy or CCS (PE, SE and FE in TWa, or CCS and OAE in Mt CO2) per target group (total, biomass, fossil, VRE, renewables, synthetic, ...).
```
add example for OAE as it's different from others?

strefler commented on main.gms:
```diff
@@ -848,6 +848,32 @@ parameter
 *' * (0): not included
 *'
+parameter
+  cm_33OAE                  "choose whether OAE (ocean alkalinity enhancement) should be included into the CDR portfolio. 0 = OAE not used, 1 = used"
```
maybe add comment that global bound should be used? with example?

strefler commented on modules/33_CDR/portfolio/datainput.gms:
```diff
@@ -49,5 +49,27 @@ p33_fedem("weathering", "fedie") = 0.3;
+
+*** ocean alkalinity enhancement input data (Kowalczyk et al., 2024)
+
+!! An assumption; generally the efficiency might vary between 1.2-1.8 tCO2/tCaO,
```
now it's been changed to 0.9-1.4

strefler commented on modules/33_CDR/portfolio/datainput.gms:
```diff
@@ -49,5 +49,27 @@ p33_fedem("weathering", "fedie") = 0.3;
+
+p33_fedem(te_oae33, "feels") = 1.0 / s33_OAE_efficiency; !! 996 MJ/tCaO
+p33_fedem(te_oae33, "fehes") = 3.1 / s33_OAE_efficiency; !! 3100 MJ/tCaO
+
```
maybe add more comments what FE is for?

strefler commented on modules/33_CDR/portfolio/equations.gms:
```diff
@@ -34,42 +34,47 @@ q33_emiCDR(t,regi)..
-    - vm_emiCdrTeDetail(t,regi,"dac")
+    - vm_emiCdrTeDetail(t, regi, "dac")
     + (1 / pm_eta_conv(t,regi,"gash2c")) * fm_dataemiglob("pegas","seh2","gash2c","cco2") * sum(fe2cdr("fegas",entyFe2,te_used33), v33_FEdemand(t,regi,"fegas", entyFe2,te_used33))
+    - s33_OAE_chem_decomposition * sum(te_oae33, vm_emiCdrTeDetail(t, regi, te_oae33))
```
add capture rate

strefler commented on modules/33_CDR/portfolio/equations.gms:
```diff
@@ -34,42 +34,47 @@ q33_emiCDR(t,regi)..
     ;
```
add emissions not captured here, so that this is the "net CDR". Would work the same way for DAC

dklein-pik committed:
Give the option to configure MAgPIE by adding individual MAgPIE switches (as defined in the default.cfg from MAgPIE) as columns to the scenario_config_coupled*.csv. The switches need to have the prefix "cfg_mag$" to be identified correctly.

dklein-pik committed:
Merge branch 'develop' of github.com:remindmodel/remind into develop

katarkow committed:
Read OAE capacity factor from input data

katarkow committed:
Update documentation and rename vm_ccs_cdr

dklein-pik committed:
Add the option to select MAgPIE scenarios from multiple scenario_config*.csv files

dklein-pik committed:
When reading the scenario_config_coupled.csv recognize columns containing 'scenario_config' as knownColumns.
Add backwards compatibility to start_bundle_coupled.R: still allow using 'magpie_scen' as name of the column that contains the MAgPIE scenarios. Use MAgPIE's 'config/scenario_config.csv' in this case.

dklein-pik committed:
Update coupling tutorial: how to specify MAgPIE scenarios using the path to a config file in the coupled config.

dklein-pik committed:
Merge branch 'develop' of github.com:remindmodel/remind into develop

dklein-pik committed:
Update 04_RunningREMINDandMAgPIE.md

dklein-pik committed:
Update 04_RunningREMINDandMAgPIE.md

dklein-pik committed:
Update coupling tutorial

dklein-pik committed:
Remove obsolete lines of code.

dbantje committed:
Updated documentation on pure rate of time preference standard value.

dklein-pik committed:
Return to the previous wording.

katarkow committed:
How to use bounds for OAE

katarkow committed:
Fix bug in emissions

katarkow committed:
Restore remind2

katarkow committed:
Clean up

katarkow committed:
Clean up fixings

katarkow committed:
Fix warnings

katarkow committed:
Remove unused fixings

dklein-pik committed:
Update tutorials/04_RunningREMINDandMAgPIE.md

Co-authored-by: Leon Merfort <58845874+merfort@users.noreply.github.com>

dklein-pik committed:
Only specify memory requirements for cs2 if not already selected by user via outputR.
Insted of --mem-per-cpu use --mem (as also done in output.R) to avoid forbidden combination of memory parameters.

dklein-pik committed:
Add --partition=priority for subsequent runs if submitted with --qos=priority

dklein-pik committed:
Filter empty fields (NA) in MAgPIE scenarios and switches.
Add --partition=priority for coupled runs if submitted with --qos=priority.
Fix typo.

dklein-pik committed:
Include the two new options to select MAgPIE scenarios in the coupling to the tests:
Add (i) column with path to an additional MAgPIE scenario_config.csv and (ii) individual MAgPIE switches to scenario_config_coupled_shortCascade.csv.

dklein-pik committed:
Merge branch 'develop' of github.com:dklein-pik/remind into develop

dklein-pik committed:
Bugfix: keep data frame even if only one column remains after filtering

dklein-pik committed:
Remove outdated scenarios from coupled config.

dklein-pik committed:
Merge pull request #1759 from dklein-pik/develop

Add two new options to configure MAgPIE in a coupled run

RSEBot committed:
Release development version 3.3.2.dev267

dklein-pik committed:
Undo the accidental activation of a coupling scenario.

dklein-pik committed:
Merge branch 'develop' of github.com:remindmodel/remind into develop

dklein-pik committed:
Merge pull request #1761 from dklein-pik/develop

Undo the accidental activation of a coupling scenario

RSEBot committed:
Release development version 3.3.2.dev271

merfort committed:
Prevent overwriting of peak budget year in loop

Renato-Rodrigues committed:
enable bioigcc early retirement

dbantje committed:
Merge pull request #1760 from dbantje/prtp_docu

Updated documentation on `cm_prtpScen` switch that sets the pure rate of time preference.

RSEBot committed:
Release development version 3.3.2.dev274

cchrisgong committed:
add nuclear endogenous switch

orichters committed:
fail tests if manipulating main.gms with default cfg drops/changes switches

orichters committed:
short filename, make sure it works also if diff is missing

cchrisgong committed:
adding an intermediate assumption option for ccsinjecratescen

cchrisgong committed:
adding an intermediate assumption option for ccsinjecratescen

cchrisgong committed:
Merge pull request #1765 from cchrisgong/addswitch

adding non-default switch value for nucscen and ccsinjecratescen

RSEBot committed:
Release development version 3.3.2.dev279

orichters committed:
Merge pull request #1764 from orichters/codecheck

fail tests if manipulating main.gms with default cfg drops/changes switches

RSEBot committed:
Release development version 3.3.2.dev283

orichters committed:
test that manipulateFile does not destroy comments. Needs lucode2#205

orichters committed:
make sclass command conditional on existence

merfort committed:
Add cm_peakBudgYr to display statement

merfort committed:
Merge pull request #1766 from merfort/bugfix__peak_budget_algorithm

Prevent overwriting of peak budget year in loop

RSEBot committed:
Release development version 3.3.2.dev287

Renato-Rodrigues committed:
Merge pull request #1763 from Renato-Rodrigues/bioigcc_earlyRetirement

Enable bioigcc early retirement

RSEBot committed:
Release development version 3.3.2.dev290

orichters committed:
remove all testthat files

orichters committed:
check manipulateConfig with arbitrary settings

orichters committed:
use also non-gms cfg$ switches in start_bundle_coupled.R

orichters committed:
Merge pull request #1768 from orichters/startcoupled

use also non-gms cfg$ switches in start_bundle_coupled.R

RSEBot committed:
Release development version 3.3.2.dev293

orichters committed:
Merge pull request #1767 from orichters/codecheck

test that manipulateConfig does not destroy comments and can undo changes

RSEBot committed:
Release development version 3.3.2.dev299

orichters committed:
add --partition=priority to --qos=standby

orichters committed:
Update xlsx_IIASA.R, load iiasatemplate from server

orichters committed:
Update DESCRIPTION: piamInterfaces >= 0.21.0

orichters committed:
Merge pull request #1772 from orichters/startcoupled

Use server link für NGFS and ELEVATE iiasatemplate

RSEBot committed:
Release development version 3.3.2.dev303

orichters committed:
Update checkProjectSummations.R - use intensiveUnits not skipUnits

orichters committed:
Update DESCRIPTION - piamInterfaces >= 0.22.0

orichters committed:
add #1773 to changelog

orichters committed:
Merge pull request #1773 from orichters/startcoupled

also check intensive variables in checkSummationsRegional

RSEBot committed:
Release development version 3.3.2.dev308

orichters committed:
undo adding --partition, problem is solved

orichters committed:
Merge pull request #1770 from orichters/codecheck

remove --partition= specification, as problem is solved

RSEBot committed:
Release development version 3.3.2.dev312

orichters committed:
adjust config files from remind-3p3p1

orichters committed:
bug: cfg is 'TRUE' if specified via config file

orichters committed:
do not remove BAU in coupled config where carbonprice is not defined

orichters committed:
integratedDamageCosts: remove data already present, generalize

laurinks committed:
bugfix in 45_carbonprice realization expoLinear

laurinks committed:
Merge pull request #1774 from laurinks/expoLinear-bugfix

bug fix in 45_carbonprice realization expoLinear

RSEBot committed:
Release development version 3.3.2.dev315

katarkow committed:
OAE documentation

katarkow committed:
Calculated emissions captured from gas and calcination separately

orichters committed:
update config files with NGFS damage runs

orichters committed:
fix compilation errors and renamed switch

orichters committed:
update iiasa submission tutorial

orichters committed:
adjust readCheckScenarioConfig to set default one cfg$ switches

orichters committed:
copy h_cpol setting to o_cba runs

orichters committed:
adapt percent->% for harmonization

katarkow committed:
Add missing space

katarkow committed:
OAE in CHANGELOG

fbenke-pik committed:
remove header from Rmd file cesCalibrationReport.R

fbenke-pik committed:
Merge pull request #1778 from fbenke-pik/develop

remove header from Rmd file cesCalibrationReport.Rmd

RSEBot committed:
Release development version 3.3.2.dev318

fbenke-pik committed:
add EDGE-T to cs2

fbenke-pik committed:
fix typo in profiles.json

laurinks committed:
Removing unused variable v01_enerSerAdj and set in_enerSerAdj from 01_macro/singleSectorGr

orichters committed:
Merge pull request #1775 from orichters/NGFSv5-clean

updates and bugfixes from NGFSv5

RSEBot committed:
Release development version 3.3.2.dev330

laurinks committed:
Merge pull request #1781 from laurinks/code-cleaning-enerSerAdj

Removing unused variable v01_enerSerAdj and set in_enerSerAdj from 01_macro/singleSectorGr

RSEBot committed:
Release development version 3.3.2.dev333

fbenke-pik committed:
update changelog

fbenke-pik committed:
Merge pull request #1780 from fbenke-pik/develop

add EDGE-T to cs2

RSEBot committed:
Release development version 3.3.2.dev338

strefler commented on core/equations.gms:
```diff
@@ -740,18 +740,19 @@ q_emiMac(t,regi,emiMac) ..
   -  p_macBaseMagpieNegCo2(t,regi)
-       !! negative emissions from the cdr module that are not stored geologically
-       -       (vm_emiCdr(t,regi,"co2") + sum(teCCS2rlf(te,rlf), vm_ccs_cdr(t,regi,"cco2","ico2","ccsinje",rlf)))
+  !! negative emissions from the cdr module that are not stored geologically
```
may want to mention that this is EW and OAE

orichters committed:
limit slurm runtime of output.R scripts to 5 hours

Commit:
remove duplicate code

Commit:
rename set to 'pese' and move declaration to core

Commit:
update ariadne scenario config

orichters committed:
folder does not yet exist, therefore avoid warning

orichters committed:
adjust output.R runtime to 2 hours

orichters committed:
Merge pull request #1783 from orichters/develop

limit slurm runtime of output.R scripts to 5 hours

RSEBot committed:
Release development version 3.3.2.dev343

Commit:
remove non-existing switches from ariadne scenario config

Commit:
move remaining 2025 bounds behind cm_tech_bounds_2025 switch

orichters committed:
check for duplicates in mif

orichters committed:
shorten slurm time for coupled run tests

orichters committed:
read withoutPlus mif, remove inf from MAgPIE

Commit:
add note to historical fixings in regipol module

Commit:
remove Germany CCS limit for 2030 which had been removed in the trunk

fschreyer committed:
Merge pull request #1714 from fschreyer/ariadne_merge_Germany

Add Germany-specific changes and scenario config from Ariadne project

RSEBot committed:
Release development version 3.3.2.dev359

Renato-Rodrigues committed:
add 2040 to the EU27 comparison scenario plots

orichters committed:
Merge pull request #1784 from orichters/develop

 check for duplicates in mif, reduce coupled test slurm time

RSEBot committed:
Release development version 3.3.2.dev364

Renato-Rodrigues committed:
Merge pull request #1785 from Renato-Rodrigues/compScen

add 2040 to the EU27 comparison scenario plots

RSEBot committed:
Release development version 3.3.2.dev367

katarkow committed:
Merge remote-tracking branch 'upstream/develop' into OAE_0db5204
